pkg/datarepair/checker/checker.go

```diff
@@ -127,8 +127,13 @@ func (checker *Checker) IdentifyInjuredSegments(ctx context.Context) (err error)
 				}
 
 				remoteSegmentsChecked++
-				numHealthy := len(pieces) - len(missingPieces)
-				if (int32(numHealthy) >= pointer.Remote.Redundancy.MinReq) && (int32(numHealthy) <= pointer.Remote.Redundancy.RepairThreshold) {
+				numHealthy := int32(len(pieces) - len(missingPieces))
+				redundancy := pointer.Remote.Redundancy
+				if numHealthy >= redundancy.MinReq && numHealthy <= redundancy.RepairThreshold && redundancy.RepairThreshold != redundancy.SuccessThreshold {
```

Comment:
any chance we can add a comment here to explain the condition where this `if` block is true, kind of a lot going on here?